### PR TITLE
Added opts.keyGenerator and used req.ip for IP address resolving

### PR DESCRIPTION
**Pull request by mcibique**
Sorry, took me a while to get back to the #7 . I have added new option `resolveIpAddress`, updated README and added tests. Let me know if there is anything unclear or can be done better. Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,16 +49,28 @@ fastify.register(require('fastify-rate-limit'), {
...
-- `whitelist`: array of string of ips to exlude from rate limiting
+- `whitelist`: array of string of ips to exclude from rate limiting.
 - `redis`: by default this plugins uses an in-memory store, which is fast but if you application works on more than one server it is useless, since the data is store locally.<br>
 You can pass a Redis client here and magically the issue is solved. To achieve the maximum speed, this plugins requires the use of [`ioredis`](https://github.com/luin/ioredis).
-- `skipOnError`: if `true` it will skip errors generated by the storage (eg, redis not reachable)
+- `skipOnError`: if `true` it will skip errors generated by the storage (eg, redis not reachable).
+- `resolveIpAddress`: a function for resolving client's IP address. Use in case your fastify server is behind the proxy. You can check for request header `X-Forwarded-For` or anything else attached to the request by your proxy, but only if you trust the provided value. Check out [#7](https://github.com/fastify/fastify-rate-limit/issues/7) how untrusted value can lead to bypassing the rate limiter. Example usage:
```
Comment (mcollina) [inline README.md]: Can you add that explanation to for that somewhere in the README?

**Comment by mcibique** [comment]
Aaah, I didn't know you have trust proxy implemented. Thanks for pointing it out. I see in the fastify [code](https://github.com/fastify/fastify/blob/master/fastify.js#L80) that if you trust the proxy, `req.ip` is determined from the `X-Forwarded-For` header, otherwise it goes for `req.connection.remoteAddress`. This plugin uses the `req.connection.remoteAddress` directly, and that's ignoring the entire logic in fastify. However, it is easy to fix it by replacing the default fallback: 
```js
const resolveIpAddress = typeof opts.resolveIpAddress === 'function'
    ? opts.resolveIpAddress
    : (req) => req.ip
```

I can update the code in PR and also update the README saying that you can use trustProxy, or if you cannot trust the proxy, you can use own `resolveIpAddress` function to determine the IP.

**Comment by mcollina** [comment]
Note that you can pass `trustProxy: function (addr) { }`, because it's based on https://www.npmjs.com/package/proxy-addr.

**Comment by mcibique** [comment]
I was looking into that function but unfortunately that doesn't solve my problem. `proxyaddr` always get the IP address from 'X-Forwarded-For' header, but I cannot use that header, because my fastify server is behind nginx proxy. Nginx is configured to append its IP to `X-Forwarded-For`, doesn't override it, so the value can be spoofed and cannot be trusted. The remote address of the client is sent from nginx in another header called `X-Real-IP`. That one cannot be spoofed and that's why I'd like to have an option to specify my own logic where to look for the IP address.

Currently I have to do this in my server code to make it working:
```
fastify.addHook('onRequest', function (req, res, next) {
  req.headers['X-Forwarded-For'] = req.headers['x-real-ip']
})
```
... and this is very ugly workaround.

**Pull request merged**